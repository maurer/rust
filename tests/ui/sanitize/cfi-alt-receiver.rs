// Check alternate receivers work

// needs-sanitizer-cfi
// compile-flags: --crate-type=bin -Cprefer-dynamic=off -Clto -Zsanitizer=cfi -C codegen-units=1 -C opt-level=0
// run-pass

// FIXME: opt-level=0 is load bearing, and shouldn't be
// FIXME: we're emitting a warning for an unexpected initial local type, need to deal with that or
// verify it happens without shims

use std::sync::Arc;

trait Fooable {
    fn foo(self: Arc<Self>);
}

struct Bar;

impl Fooable for Bar {
    fn foo(self: Arc<Self>) {}
}

fn main() {
    let bar: Arc<dyn Fooable> = Arc::new(Bar);
    bar.foo();
}
