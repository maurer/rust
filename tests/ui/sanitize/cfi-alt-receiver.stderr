WARN rustc_codegen_ssa::mir::locals Unexpected initial operand type: expected std::sync::Arc<dyn [Binder(Trait(Fooable), [])] + ReErased, std::alloc::Global>, found *const alloc::sync::ArcInner<dyn [Binder(Trait(Fooable), [])] + ReErased>.See <https://github.com/rust-lang/rust/issues/114858>.
